### DISCUSSION TITLE
GNNExplainer for edge regression task

### DISCUSSION THREAD

**Discussion by antoni4040** [pyg-team/pytorch_geometric#4165] [Q&A]
Hello. Can GNNExplainer be used for edge regression tasks? Or is it only for node prediction and graph prediction?

**Comment by rusty1s** [comment]
`GNNExplainer` currently only works for node- and graph-level tasks. Our newly added `to_captum` functionality works for all tasks though, see [this discussion](https://github.com/pyg-team/pytorch_geometric/discussions/4058).

**Reply by antoni4040** [reply]
How do I import to_captum? I can't import it from nn or nn.models.

**Reply by rusty1s** [reply]
You need to install from master:
```
pip install git+https://github.com/pyg-team/pytorch_geometric.git
```
It just got recently added to PyG.

**Reply by antoni4040** [reply]
I'm trying to do something like this:
```
edge_index = 1
captum_model = to_captum(model, mask_type='node') 
ig = IntegratedGradients(captum_model)
data1 = geometricData[0]
ig_attr_node = ig.attribute(data1)
```

And the model's forward function starts like this:
```
def forward(self, data):
        x, edge_index = data.x, data.edge_index
        x = self.first(x).relu()
        x = self.second(x).relu()
        x = self.third(x)
```
And I get:
```
AssertionError: `inputs` must have type torch.Tensor but <class 'torch_geometric.data.data.Data'> found: 
```
Do I need to change it to accept x, edge_index instead of the bundled Data object?

**Reply by rusty1s** [reply]
Yes, this is indeed necessary. See [here](https://github.com/pyg-team/pytorch_geometric/blob/master/examples/captum_explainability.py) for an example.
